### PR TITLE
MAHOUT-1888: Performance Bug with Mahout Vector Serialization

### PR DESCRIPTION
**Pull request by smarthi**

### CODE REVIEW

**Comment by smarthi** [comment]
Do not merge, this is still WIP.

File: spark/src/test/scala/org/apache/mahout/sparkbindings/blas/BlasSuite.scala [hunk 1]
```diff
@@ -39,11 +42,11 @@ class BlasSuite extends FunSuite with DistributedSparkSuite {
     val drmA = drmParallelize(m = inCoreA, numPartitions = 3)
     val drmB = drmParallelize(m = inCoreB, numPartitions = 2)
 
-    val op = new OpABt(drmA, drmB)
+    val op = OpABt(drmA, drmB)
 
     val drm = new CheckpointedDrmSpark(ABt.abt(op, srcA = drmA, srcB = drmB), op.nrow, op.ncol)
 
-    printf("AB' num partitions = %d.\n", drm.rdd.partitions.size)
+    printf("AB' num partitions = %d.\n", drm.rdd.partitions.length)
```
Comment (dlyubimov) [inline spark/src/test/scala/org/apache/mahout/sparkbindings/blas/BlasSuite.scala]: in scala, we use size() to measure collection cardinality. using .length may cause conversion to a java collection.

File: spark/src/test/scala/org/apache/mahout/sparkbindings/test/DistributedSparkSuite.scala [hunk 2]
```diff
@@ -45,6 +45,7 @@ trait DistributedSparkSuite extends DistributedMahoutSuite with LoggerConfigurat
           .set("spark.akka.frameSize", "30")
           .set("spark.default.parallelism", "10")
           .set("spark.executor.memory", "2G")
+          .set("spark.kryo.registrationRequired", "true")
```
Comment (dlyubimov) [inline spark/src/test/scala/org/apache/mahout/sparkbindings/test/DistributedSparkSuite.scala]: This is not needed, this is why it is failing. we can enable that to see what else is left, but we don't have to patch every class out there that is used in tests. I think the classes that still have that have something to do with IndexedDataSet which is (in my view) not part of algebra engine, so we can ignore the rest.

Comment (smarthi) [inline spark/src/test/scala/org/apache/mahout/sparkbindings/test/DistributedSparkSuite.scala]: Fixed the comments.

Comment (dlyubimov) [inline spark/src/test/scala/org/apache/mahout/sparkbindings/test/DistributedSparkSuite.scala]: please remove line 48

Comment (smarthi) [inline spark/src/test/scala/org/apache/mahout/sparkbindings/test/DistributedSparkSuite.scala]: Its been removed, the recent github changes for code review retain the original lines that were commented; confused me too.

**Pull request closed without merging**